List messages

lopesnuno opened the pull request.

lopesnuno committed:
List messages

lopesnuno committed:
- Fixed syntax;
- Changed promise type on 'MessageService'

kimile commented on src/modules/message/MessageRepository.ts:
```diff
@@ -29,23 +29,43 @@ export default class MessageRepository implements Repository<Message> {
+      connection.query(sql`
+          SELECT *
+          FROM messages
+          WHERE rooms_id = ${id} LIMIT 50;
```
Este campo não se devia chamar `rooms_id`, mas sim `room_id`, sendo consistente com as outras foreign keys na base de dados... favor alterar!

kimile commented on src/modules/message/MessageRepository.ts:
```diff
@@ -29,23 +29,43 @@ export default class MessageRepository implements Repository<Message> {
+      const obj = rows[i];
 
-        return rowCount === 1;
+      messages.push(new Message(obj.id, obj.message, obj.sender_id, obj.recipient_id, obj.reply_to, obj.rooms_id, new Date(obj.send_at as number)));
```
O campo `send_at` devia ser `sent_at` para traduzir correctamente para "enviada a..." e não "a enviar a..."

kimile commented on src/modules/message/MessageApi.ts:
```diff
@@ -61,15 +75,16 @@ const deleteMessage: RequestHandler = async (req: Request, res: Response, next:
 
 export default (app: Router): void => {
-  app.post('/message/', Auth.authorize([]), create);
+    app.get('/message/:id', Auth.authorize([]), list);
```
Foi isto que tínhamos discutido para o endpoint deste request? Não foi `/messages/:id`? Como é que vai ficar o endpoint para pedir os detalhes de uma mensagem específica (exemplo, ver quando foi lida)?

kimile replied:
Outra nota - era melhor chamar àquele `:id` algo mais claro como `room`: tive de ir até ao repository para saber o que aquilo era

lopesnuno committed:
# Changed list endpoint => /messages/:room
# Changed db column name rooms_id => room_id and send_at => sent_at
# Changed MessageModel.ts sendAt => sentAt
# Added room to fixtures.sql and syntax fixes

lopesnuno committed:
Message list order by date => DESC
